### PR TITLE
SOV-439: WalletConnect

### PR DESCRIPTION
**Pull request by creed-victor**
https://sovryn.atlassian.net/browse/SOV-439

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,12 +18,13 @@
...
+    "commitlint": "^17.1.2",
     "eslint-config-custom": "*",
     "husky": "^8.0.0",
     "lint-staged": "^13.0.3",
-    "typescript": "^4.5.2",
-    "prettier": "latest",
-    "turbo": "latest"
+    "prettier": "^2.7.1",
+    "turbo": "latest",
+    "typescript": "^4.5.2"
```
Comment (tiltom) [inline package.json:27]: Delete  `^`, see https://sovryn.monday.com/docs/2733202752?blockId=aebe8f4b-5e4a-4ef6-8967-503ce7a1bb30 .

File: packages/walletconnect/package.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "typescript": "^4.5.2"
+  },
+  "dependencies": {
+    "@ethersproject/providers": "^5.7.1",
+    "@sovryn/onboard-common": "*",
+    "@sovryn/onboard-hw-common": "*",
+    "@walletconnect/client": "^1.8.0",
+    "@walletconnect/qrcode-modal": "^1.8.0",
+    "rxjs": "^7.5.6"
+  }
```
Comment (tiltom) [inline packages/walletconnect/package.json:28]: Delete `^`, see https://sovryn.monday.com/docs/2733202752?blockId=aebe8f4b-5e4a-4ef6-8967-503ce7a1bb30 .

File: packages/walletconnect/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+
+        const { ProviderRpcError, ProviderRpcErrorCode } = await import(
+          "@sovryn/onboard-common"
+        );
+
+        const { default: WalletConnect } = await import(
+          "@walletconnect/client"
+        );
+
+        // This is a cjs module and therefor depending on build tooling
```
Comment (tiltom) [inline packages/walletconnect/src/index.ts:42]:
```suggestion
        // This is a cjs module and therefore depending on build tooling
```

**Review by tiltom — CHANGES_REQUESTED** [review]

**Review by tiltom — CHANGES_REQUESTED** [review]

**Review by tiltom — APPROVED** [review]

**Comment by soulBit** [comment]
Tested with Argent and Trust wallets on Android 12

Argent:
* connects successfully
* fails to sign message:
```
{
    "address": "0x9733b57e33046f4e1f8ed88f433c6364d126b4e0",
    "signer": "0x0d3ad0538Eb10E2207C5A8822427327fe3C7948B",
    "signature": "0x68994d682674b70d3dfc3d167cbf905a0f51d5846878ca1f67b7715dffc878300de89cd57a18eff506c45a5d85145a4998dce91e35e547ba90f401667702e61e1c"
}
```
* fails to sign typed data:
```
{
    "address": "0x9733b57e33046f4e1f8ed88f433c6364d126b4e0",
    "signer": "0x0d3ad0538Eb10E2207C5A8822427327fe3C7948B",
    "signature": "0x27d7016f0ff0bfbb9304764ca94d437653145b40fd9e25e2a1719ac95214da313c6f181ec2c1b6ca396a3d4d73c78a35040202cab16730c6c535a0bc7d2e51331b"
}
```

Trust wallet:
* does not connect

**Review by soulBit — CHANGES_REQUESTED** [review]
Tested with Argent and Trust wallets on Android 12

Argent:
* connects successfully
* fails to sign message:
```
{
    "address": "0x9733b57e33046f4e1f8ed88f433c6364d126b4e0",
    "signer": "0x0d3ad0538Eb10E2207C5A8822427327fe3C7948B",
    "signature": "0x68994d682674b70d3dfc3d167cbf905a0f51d5846878ca1f67b7715dffc878300de89cd57a18eff506c45a5d85145a4998dce91e35e547ba90f401667702e61e1c"
}
```

* fails to sign typed data:
```
{
    "address": "0x9733b57e33046f4e1f8ed88f433c6364d126b4e0",
    "signer": "0x0d3ad0538Eb10E2207C5A8822427327fe3C7948B",
    "signature": "0x27d7016f0ff0bfbb9304764ca94d437653145b40fd9e25e2a1719ac95214da313c6f181ec2c1b6ca396a3d4d73c78a35040202cab16730c6c535a0bc7d2e51331b"
}
```

Trust wallet:

* does not connect

**Comment by soulBit** [comment]
Defiant, Metamask mobile working well - we need to define exactly which WalletConnect wallets work later, but approving for now to unblock development

**Review by soulBit — APPROVED** [review]

**Pull request merged**